mmcls/core/hook/NumClassCheckHook.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) OpenMMLab. All rights reserved
+from mmcv.runner.hooks import Hook
+
+from .builder import HOOKS
+
+
+@HOOKS.register_module()
+class NumClassCheckHook(Hook):
+
+    def _check_head(self, runner):
+        """Check whether the `num_classes` in head matches the length of
+        `CLASSES` in `dataset`.
+
+        Args:
+            runner (obj:`EpochBasedRunner`): Epoch based Runner.
+        """
+        model = runner.model
+        dataset = runner.data_loader.dataset
+        if dataset.CLASSES is None:
+            runner.logger.warning(
+                f'Please set `CLASSES` '
+                f'in the {dataset.__class__.__name__} and'
+                f'check if it is consistent with the `num_classes` '
+                f'of head')
+        else:
+            assert type(dataset.CLASSES) is not str, \
+                (f'`CLASSES` in {dataset.__class__.__name__}'
+                 f'should be a tuple of str.'
+                 f'Add comma if number of classes is 1 as '
+                 f'CLASSES = ({dataset.CLASSES},)')
```

Comment:
You can check it with:
```python
from mmcv.utils import is_seq_of
...
assert is_seq_of(dataset.CLASSES, str)
```
And the suggestion is not necessary, I think it's uncommon that the number of classes is one.